sim/pokemon.ts

```diff
@@ -1096,8 +1066,8 @@ export class Pokemon {
 			for (const typeid of type) {
 				if (this.hasType(typeid)) return true;
 			}
-		} else {
-			if (this.getTypes().includes(type)) return true;
+		} else if (this.getTypes().includes(type)) {
+			return true;
```

Comment:
I think the previous layout is more readable: Do one check if `type` is an array, do the other check if `type` isn't an array.

This is the sort of thing that I think we'd lose by switching to Prettier, incidentally.